src/Microsoft.DotNet.Build.Tasks/PackageFiles/buildagainstpackages.targets

```diff
@@ -0,0 +1,181 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="12.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <UsingTask TaskName="AddDependenciesToProjectJson" AssemblyFile="$(BuildToolsTaskDir)Microsoft.DotNet.Build.Tasks.dll"/>
+  <UsingTask TaskName="ParsePackageNames" AssemblyFile="$(BuildToolsTaskDir)Microsoft.DotNet.Build.Tasks.dll"/>
+  <UsingTask TaskName="GetPackageNumberFromPackageDrop" AssemblyFile="$(BuildToolsTaskDir)Microsoft.DotNet.Build.Tasks.dll"/>
+
+  <ItemGroup>
+    <_PackagesDrops Include="$(PackagesDrops)" />
+  </ItemGroup>
+  <PropertyGroup>
+    <!-- overridden in repo -->
+    <GeneratedProjectJsonDir Condition="'$(GeneratedProjectJsonDir)' == ''">$(MSBuildThisFileDirectory)generated</GeneratedProjectJsonDir>
```

Comment:
To be clear I don't like this default because it will be writing in the tool directory. 

---
In reply to: [58455919](https://github.com/dotnet/buildtools/pull/571#discussion_r58455919) [](ancestors = 58455919)